app/main/forms.py

```diff
@@ -2563,3 +2580,23 @@ class PlatformAdminSearch(StripWhitespaceForm):
         param_extensions={"hint": {"text": ("Search for users, services, and organisations by name or partial name.")}},
         validators=[DataRequired()],
     )
+
+
+class PostcodeForm(StripWhitespaceForm):
+    postcode = PostcodeSearchField("Postcode")
+    radius = GovukDecimalField(
+        "Add radius",
+        param_extensions={
+            "classes": "govuk-input govuk-input--width-5",
+            "suffix": {"text": "km"},
+            "attributes": {"pattern": "[0-9]*"},
+        },
+        validators=[
+            NumberRange(min=0.099, max=38.001, message="Enter a radius between 0.1km and 38.0km."),
+            DataRequired(message="Enter a radius."),
+            Only2DecimalPlaces(),
```

Comment:
I don't think you can enter with decimal places, since up above the attributes pattern above doesn't allow a decimal point?